src/main/java/mat/server/service/FhirValidationReportService.java

```diff
@@ -95,31 +101,79 @@ private String generateValidationReport(ConversionResultDto conversionResultDto,
         paramsMap.put("matVersion", currentMatVersion);
 
         if (conversionResultDto != null && measure != null) {
-            //Measure FHIR validation errors
-            List<FhirValidationResult> measureFhirValidationErrors = conversionResultDto.
-                    getMeasureConversionResults().getMeasureFhirValidationResults();
-
-            // Library FHIR validation errors
-            List<LibraryConversionResults> libraryConversionResults = conversionResultDto.getLibraryConversionResults();
-            List<FhirValidationResult> libraryFhirValidationErrors = libraryConversionResults.stream().flatMap(i -> i.getLibraryFhirValidationResults().stream()).collect(Collectors.toList());
-
-            // CQL conversion errors
-            List<CqlConversionError> cqlConversionErrors = libraryConversionResults.stream().map(i -> i.getCqlConversionResult()).flatMap(i -> i.getCqlConversionErrors().stream()).collect(Collectors.toList());
-
-            // ValueSet FHIR validation errors
-            List<FhirValidationResult> valueSetFhirValidationErrors = conversionResultDto.getValueSetConversionResults().stream().flatMap(i -> i.getValueSetFhirValidationResults().stream()).collect(Collectors.toList());
+            if (!("SUCCESS").equals(conversionResultDto.getOutcome().name())) {
+
+
+                if(!conversionResultDto.getErrorReason().isEmpty()) {
+                    paramsMap.put("ErrorReason", conversionResultDto.getErrorReason());
+                }
+                // ValueSet FHIR validation errors
+                List<FhirValidationResult> valueSetFhirValidationErrors = new ArrayList<>();
+                if (CollectionUtils.isNotEmpty(conversionResultDto.getValueSetConversionResults())) {
+                    for (ValueSetConversionResults valueSetConversionResults : conversionResultDto.getValueSetConversionResults()) {
+                        if (Boolean.FALSE.equals(valueSetConversionResults.getSuccess())) {
```

Comment:
I recall in some cases we get nulls. Not sure if it's a success or a failure..

Reply:
I discussed with Duane on all possible null occurrences, As of our Knowledge, I handled all of them.